octopusdeploy/schema_kubernetes_agent_worker.go

```diff
@@ -0,0 +1,157 @@
+		},
+		"machine_policy_id": {
+			Description: "Optional ID of the machine policy that the Kubernetes agent will use. If not provided the default machine policy will be used.",
+			Computed:    true,
+			Optional:    true,
+			Type:        schema.TypeString,
+		},
+		"thumbprint": {
+			Description: "The thumbprint of the Kubernetes agent's certificate used by server to verify the identity of the agent. This is the same thumbprint that was used when installing the agent.",
+			Required:    true,
+			Type:        schema.TypeString,
+		},
+		"uri": {
+			Description: "The URI of the Kubernetes agent's used by the server to queue messages. This is the same subscription uri that was used when installing the agent.",
+			Required:    true,
+			Type:        schema.TypeString,
+		},
+		"upgrade_locked": {
+			Description: "If enabled the Kubernetes agent will not automatically upgrade and will stay on the currently installed version, even if the associated machine policy is configured to automatically upgrade.",
+			Optional:    true,
+			Type:        schema.TypeBool,
+		},
+		"is_disabled": {
+			Description: "Whether the Kubernetes agent is disabled. If the agent is disabled, it will not be included in any deployments.",
+			Optional:    true,
+			Default:     false,
+			Type:        schema.TypeBool,
+		},
+		"worker_pool_ids": {
+			Description: "A list of worker pool Ids specifying the pools in which this worker belongs",
+			Elem:        &schema.Schema{Type: schema.TypeString},
+			MinItems:    1,
+			Required:    true,
+			Type:        schema.TypeList,
+		},
+
+		// Read-only Values
+		"agent_version": {
+			Description: "Current Helm chart version of the agent.",
+			Computed:    true,
+			Type:        schema.TypeString,
+		},
+		"agent_tentacle_version": {
+			Description: "Current Tentacle version of the agent",
+			Computed:    true,
+			Type:        schema.TypeString,
+		},
+		"agent_upgrade_status": {
+			Description: "Current upgrade availability status of the agent. One of 'NoUpgrades', 'UpgradeAvailable', 'UpgradeSuggested', 'UpgradeRequired'",
+			Computed:    true,
+			Type:        schema.TypeString,
+		},
+		"agent_helm_release_name": {
+			Description: "Name of the Helm release that the agent belongs to.",
+			Computed:    true,
+			Type:        schema.TypeString,
+		},
+		"agent_kubernetes_namespace": {
+			Description: "Name of the Kubernetes namespace where the agent is installed.",
+			Computed:    true,
+			Type:        schema.TypeString,
+		},
+	}
+}
+
+func getKubernetesAgentWorkerDataSchema() map[string]*schema.Schema {
+	dataSchema := getKubernetesAgentWorkerSchema()
+	setDataSchema(&dataSchema)
+
+	WorkerDataSchema := getWorkerDataSchema()
+	WorkerDataSchema["kubernetes_agent_workers"] = &schema.Schema{
+		Computed:    true,
+		Description: "A list of kubernetes agent workers that match the filter(s).",
+		Elem:        &schema.Resource{Schema: dataSchema},
+		Optional:    true,
```

Comment:
There was a recent issue around deferring the reads for datasources. This only applies to the return list, TLDR is that optional should be false here. 

There's more details on this [PR](https://github.com/OctopusDeployLabs/terraform-provider-octopusdeploy/pull/706/files#diff-e8b16e3778ca4cfdfdeba66488826a7c376db216dd027c19e689545ef3cccc9b)

Reply:
Thanks for the heads up!